docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+- Result：
+  - The ConfigMaps and Secrets that the resource depends on will also be promoted to be managed by Karmada.
+
+## Design Details
+
+### Architecture
+
+![architecture](statics/architecture.drawio.svg)
+From the above architecture diagram, the workflow for promoting the resource and its dependent resources is as follows:
+1. Retrieve the ObjInfo(Object Information) of the workload to be promoted from the member cluster.
+2. Determine whether to enable the functionality of automatically promoting all dependent resources based on the value of `Deps`. 
+   1. If `Deps` is set to `false`, only create the corresponding Resource Template, ClusterPropagationPolicy/PropagationPolicy, and generate the respective ResourceBinding in the Karmada control plane for the workload.
+   2. If `Deps` is set to `true`, use the interpret framework to analyze the dependencies of the corresponding workload. Create the Resource Templates, ClusterPropagationPolicy/PropagationPolicy, and ResourceBindings for the dependencies in the Karmada control plane. Then, create the ResourceBinding for the workload itself. 
+
+**Note:**
+If the workload to be promoted is a CR(Custom Resource), we need to first promote its CRD (Custom Resource Definition).
+
+### Interpret Workload Dependencies
+
+To achieve the automatic promotion of dependent resources, the key challenge is to determine how to obtain the dependent resources of the resource to be promoted. Therefore, we need to be able to parse the dependencies of a resource. Currently, the interpret framework in Karmada supports the following methods for parsing dependent resources:
+
+1. Default built-in interpretation
+   1. Native Kubernetes Resource Interpretation:
+      1. Perform dependency parsing using the structure of native Kubernetes resources.
+   2. Third-Party Resource Interpretation:
+      1. Pre-configure Lua scripts for commonly used CR(Custom Resource) to enable dependency parsing using Lua configurations.
+3. Custom Interpretation
+   1. Lua Configuration-based Interpretation:
+      1. Configure resource interpretation rules for dependency parsing by specifying Lua scripts in the [ResourceInterpreterCustomization](https://github.com/karmada-io/karmada/blob/84b971a501ba82c53a5ad455c2fe84d842cd7d4e/pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go#L17) resource.
+   2. Webhook-based Interpretation:
+      1. Users can design Resource Interpreter Webhooks for specific custom resources to handle Interpreter Operations requests and perform dependency parsing. The process involves receiving these request operations and processing them accordingly.
+         The specific process is as follows:
+
+![webhook-interpret](statics/webhook-interpret.png)
+
+***Note:*** 
+Using the webhook-based interpretation method may pose difficulties when making requests to the webhook interpret service. In such cases, we need to perform port forwarding to invoke the service within the cluster.
```

Comment:
> we need to perform port forwarding to invoke the service within the cluster.

Is it possible to access it outside the cluster?

Reply:
I think it's possible, and easy to implement. Just request the target url.